contracts/signature-drop/SignatureDrop.sol

```diff
@@ -202,50 +188,31 @@ contract SignatureDrop is
     //////////////////////////////////////////////////////////////*/
 
     /// @dev Claim lazy minted tokens via signature.
-    function mintWithSignature(ISignatureMintERC721.MintRequest calldata _req, bytes calldata _signature)
+    function mintWithSignature(MintRequest calldata _req, bytes calldata _signature)
         external
         payable
-        nonReentrant
         returns (address signer)
     {
-        signer = ISignatureMintERC721(sigMint).mintWithSignature(_req, _signature);
+        require(_req.quantity > 0, "minting zero tokens");
 
         uint256 tokenIdToMint = _currentIndex;
         require(tokenIdToMint + _req.quantity <= nextTokenIdToMint, "not enough minted tokens.");
 
+        // Verify and process payload.
+        signer = _processRequest(_req, _signature);
+
         // Get receiver of tokens.
-        address receiver = _req.to == address(0) ? msg.sender : _req.to;
+        address receiver = _req.to == address(0) ? _msgSender() : _req.to;
```

Comment:
we should probably inform people that this is vulnerable to signature frontrunning attack